### PR TITLE
HPCC-20553 New defects reported by Coverity Scan for HPCC-Platform on…

### PR DESCRIPTION
**Pull request by RussWhitehead**
… 17th of September

CSecureSmartSocketFactory::connect_timeout is calling secure_connect without
checking the return code.  This PR adds a check, and throws an error if
it returns a negative value

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by RussWhitehead** [comment]
@mckellyln Please review

**Comment by RussWhitehead** [comment]
@mckellyln Please review

**Comment by mckellyln** [comment]
@RussWhitehead looks fine.  There are also two cases in testsocket.cpp where this also happens.
Should those be fixed in this PR as well ?

File: system/security/securesocket/securesocket.cpp [hunk 1]
```diff
@@ -1715,7 +1715,9 @@ class CSecureSmartSocketFactory : public CSmartSocketFactory
         {
             ssock.setown(secureContext->createSecureSocket(sock.getClear()));
             // secure_connect may also DBGLOG() errors ...
-            ssock->secure_connect();
+            int res = ssock->secure_connect();
+            if (res < 0)
+                throw MakeStringException(-1, "connect_timeout : Failed to establish secure connection");
```
Comment (mckellyln) [inline system/security/securesocket/securesocket.cpp]: ok, should "connect_timeout" be changed to "secure_connect" here and below ?

**Comment by mckellyln** [comment]
@RussWhitehead just one question about exception string using "connect_timeout"

**Comment by RussWhitehead** [comment]
@mckellyln The method name in the throw identifies the method that called secure_connect. Corrected it in testsocket, please rereview

**Review by mckellyln — APPROVED** [review]
@RussWhitehead approved

**Comment by mckellyln** [comment]
@russ good to merge

**Comment by richardkchapman** [comment]
@RussWhitehead Please squash

**Comment by RussWhitehead** [comment]
@richardkchapman Squashed, please merge

**Pull request merged**